### ISSUE TITLE
[BUG] seLinux Being Disabled by Default Causing Logging Upgrade to Fail

### ISSUE THREAD

**Issue by nickwsuse** [rancher/rancher#38566]
**Rancher Server Setup**

- Rancher version: `2.6.7-rc6`
- Installation option (Docker install/Helm Chart): HELM
  - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE2
- Proxy/Cert Details: byo-valid

**Information about the Cluster**
- Kubernetes version: `v1.23.7`
- Cluster Type (Local/Downstream): `Downstream Imported AWS RKE2 cluster`

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom): `admin`

**Describe the bug**
After upgrading rancher from 2.6.6. to 2.6.7-rc6, I upgraded logging to 3.17.7; The rancher-logging app on the installed apps page shows a FAILED tag. the rancher-logging-rke2-journalId-aggregators are in crashloop and after investigation it's because the seLinux value in the values.yaml file is set to `false` when it should be set to `true`

**To Reproduce**
1. Install previous version of logging in rke2 cluster
2. configure cluster output
3. configure cluster flow
4. verify that logs are flowing to output (eg, elasticsearch)
5. upgrade logging to latest 100.1.3-up3.17.7 [pointed charts repo to dev-2.6]
6. click the "Edit YAML" button on the second page of the install
7. search for `global` > `seLinux` > `enabled` > see it's `false` instead of `true`

Result
Upgrade fails; rke2 logging aggregator pods in crashloop

Expected Result
Upgrade succeeds and logs resume flowing to elasticsearch

**Screenshots**

App Failed State:
![app_failed_state](https://user-images.githubusercontent.com/107147272/183769953-c4ff69e9-339f-4a08-97aa-1eb1c0af9427.png)

Crashloop:
![crashloop](https://user-images.githubusercontent.com/107147272/183771323-1504fbd2-fc5c-4ca3-8f80-225a39c3b8db.png)

seLinux Set to False:
![seLinux_false](https://user-images.githubusercontent.com/107147272/183771354-ae72ae24-c53a-49f2-90f7-5e0177c072f2.png)

Deployed After Setting seLinux to True:
![deployed_after_edit](https://user-images.githubusercontent.com/107147272/183771427-dbf115a1-60b3-4e63-a9dd-c6ab27103810.png)

**Comment by gaktive** [comment]
@nickwsuse ok, keep me in the loop and we can put this back into Dashboard if @bmdepesa can account for why UI since this sounds like a chart setting.

**Comment by bmdepesa** [comment]
@gaktive the problem is that the value set during the initial install is not persisted when upgrading. As we understand,  this is not part of the chart itself.

**Comment by gaktive** [comment]
@bmdepesa just so I understand properly, at some point, the `seLinux` value is changed from the earlier chart default of `false` to `true`, and then upon upgrade, despite that value being `false` in `values.yaml` in the updated chart, it should remain `true`?

Based on the linked issue, there's this comment: https://github.com/rancher/rancher/issues/38519#issuecomment-1208364910
> Upon upgrading the chart the value was reset to false causing a permission error with the aggregator when it was trying to access a privileged directory.

@catherineluse do you know where this could happen in the UI?

**Comment by catherineluse** [comment]
Yes, in that case it is a UI issue. There is UI code that attempts to take all the non-default values from the existing chart version, then pre-apply them to the values.yaml when you go to upgrade the chart, because it assumes you would want to persist all the changes you made before. (https://github.com/rancher/dashboard/blob/master/shell/pages/c/_cluster/apps/charts/install.vue#L242)

**Comment by catherineluse** [comment]
The logic for how this works in the UI is pretty generic though. I wonder if the formatting or structure of the values.yaml changed between chart versions. If so, the UI might not be able to accurately apply the same value from the old values.yaml into the new values.yaml.

**Comment by gaktive** [comment]
Thanks for the all the clarification. Transferring back to dashboard, though we need to assess if we can quickly get this in for 2.6.7.
